base/version.jl

```diff
@@ -212,14 +212,15 @@ nextminor(v::VersionNumber) = v < thisminor(v) ? thisminor(v) : VersionNumber(v.
 nextmajor(v::VersionNumber) = v < thismajor(v) ? thismajor(v) : VersionNumber(v.major+1, 0, 0)
 
 function check_new_version(existing::Vector{VersionNumber}, ver::VersionNumber)
-    @assert issorted(existing)
     if isempty(existing)
-        for v in [v"0", v"0.0.1", v"0.1", v"1"]
+        for v in [v"0.0.1", v"0.1", v"1"]
             lowerbound(v) <= ver <= v && return
         end
-        error("$ver is not a valid initial version (try 0.0.0, 0.0.1, 0.1 or 1.0)")
+        error("$ver is not a valid initial version (try 0.0.1, 0.1 or 1.0)")
     end
+    issorted(existing) || (existing = sort!(existing))
     idx = searchsortedlast(existing, ver)
+    idx > 0 || error("version $ver less than least existing version $(existing[1])")
```

Comment:
Perhaps an `ArgumentError`, since this is a condition pertaining directly to the arguments of the function?

Reply:
Perhaps but aren't all of the errors to this function argument errors?

Reply:
Yeah, I just figured if we're adding a new error condition, it would make sense to use a typed error. I would be in favor of making the other errors typed as well.

Reply:
You know what, I've changed my mind—this is not an argument error: an argument error would indicate that someone is using the function incorrectly, but the entire purpose of this function is to check if a version is valid or not. So this error does not indicate that the person is using the function incorrectly, but rather indicates that the version is not a valid version. Arguably, it should be reframed as a predicate but using an error is more convenient. The other case to be made is that whatever code calls this may want to throw an argument error...

Reply:
Okay ¯\\\_(ツ)\_/¯